numba_typing/ast_example.py

```diff
@@ -0,0 +1,54 @@
+import ast
+import inspect
+
+
+def main():
+    with open('test_generics.py') as f:
+        tree = ast.parse(f.read())
+    analyzer = Analyzer()
+    analyzer.visit(tree)
+    print(analyzer.types)
+    print(analyzer.import_from)
+
+
+class Analyzer(ast.NodeVisitor):
+    def __init__(self):
+        self.types = {}
+        self.import_from = []
+        self.module = None
+        self.function_flag = False
+
+    def visit_ImportFrom(self, node):  # need a check "import module"
+        exec(f'import {node.module}', globals())
+        self.module = node.module
+        for alias in node.names:
+            self.import_from.append(eval(f'{node.module}.{alias.name}'))
+
+    def visit_FunctionDef(self, obj):
+        self.function_flag = True
+        for i in obj.body:
+            self.visit(i)
+
+    def visit_AnnAssign(self, node):
+        if self.function_flag:
+            # need to check for child functions
+            if isinstance(node.annotation, ast.Subscript):  # containers and generics
+                container_name = node.annotation.value.id
```

Comment:
Maybe to create separate short variables for `node.annotation` and `node.target` because they're used many times. E.g. `annotation, target = node.annotation, node.target`.